### DISCUSSION TITLE
TypeScript weird issue when using a controlled input using useController from react-hook-form with an array field value

### DISCUSSION THREAD

**Discussion by cbdeveloper** [react-hook-form/react-hook-form#7844] [Q&A] [answered]
I'm building a re-usable controlled (custom select) input using `react-hook-form`.

[Example on CodeSandbox](https://codesandbox.io/s/brave-banzai-xw46ti)

They also have some examples on the `react-hook-form` official [docs](https://react-hook-form.com/api/usecontroller)

I'm getting some unexpected TypeScript behavior.

In my case, I need the field type to be `Array<string>` because this is going to be a multiple selection input. But when I do it, I get the type of the `value` (from `useController`) prop as being `string | string[]` instead of just `string[]`, which is what I would expect it to be.

[![enter image description here][1]][1]

If I just choose the type `number`, or just `string` for example. Then I get the expected types on the `value` prop:

[![enter image description here][2]][2]

[![enter image description here][3]][3]

What is happening here?

Why the `value` is being evaluated as `string | string[]` instead of just `string[]` like I'm defining in the `FieldValue` type?

  [1]: https://i.stack.imgur.com/m3bug.png
  [2]: https://i.stack.imgur.com/ZkjVm.png
  [3]: https://i.stack.imgur.com/rvBfM.png

**Comment by Moshyfawn** [comment, marked as answer]
In your example you want to pass a field name as a second generic into the `UseControllerProps` type to infer the field type
```ts
interface FormValues {
  field: string[]
}

React.FC<UseControllerProps<FormValues, "field">>
```
